keyboard/src/modules/Keyboard.rb

```diff
@@ -1504,6 +1483,43 @@ def write_udev_rule
     end
   end
 
+  # Checks if the graphical environment is being executed remotely using
+  # "ssh -X"
+  def x11_over_ssh?
+    display = Builtins.getenv("DISPLAY") || ""
+    display.split(":")[1].to_i >= 10
+  end
+
+  # Checks if it's running is text mode (no X11)
+  def textmode?
+    textmode = Linuxrc.text
+    if !Stage.initial || Mode.live_installation
+      display_info = UI.GetDisplayInfo
+      textmode = display_info && display_info["TextMode"]
+    end
+    # Ensure it's boolean
+    !!textmode
+  end
+
+  # Executes the command to set the keyboard in X11, reporting
+  # any error to the user
+  def execute_xkb_cmd
+    log.info "Setting X11 keyboard to: <#{@current_kbd}>"
+    log.info "Setting X11 keyboard:\n #{@xkb_cmd}"
+    if SCR.Execute(path(".target.bash"), @xkb_cmd) != 0
+      log.error "Failed to execute the command"
+      Popup::Error(_("Failed to set X11 keyboard to '%s'") % @current_kbd)
+    end
+  end
+
+  # Enables autorepeat if needed
+  def enable_autorepeat
+    return nil unless Stage.initial && !Mode.live_installation && !xen_running
+    cmd = "xset r on"
+    log.info "calling xset to fix autorepeat problem: #{cmd}"
```

Comment:
You have forgotten to mention: `bnc#371756: enable autorepeat`

Nobody would know which autorepeat problem it was in the past and thus would have no idea if that is still needed at all